### ISSUE TITLE
External HDMI monitor issues

### ISSUE THREAD

**Issue by FoxtrotFaux** [swaywm/sway#5261]
* Sway version: 1.4  

* Debug Log: [https://gist.github.com/FoxtrotFaux/47a48842c57947faf60c0884557d0dd6](https://gist.github.com/FoxtrotFaux/47a48842c57947faf60c0884557d0dd6)  

* Steps to reproduce:  

I have an HDMI monitor that is hooked up since boot. No problems arise other than a failure to auto-detect subpixel rendering type and optimal resolution (usually after I restart sway when the monitor is unresponsive). If I manually turn off the monitor, then turn it back on, it will display a static image (usually the last thing that was displayed on the monitor) but will be unresponsive. Sway reacts as if the monitor is still there. I can move the mouse or windows to the monitor, but they aren't shown. I tested a video benchmark out of curiosity on the monitor when it was stuck like this, and it reported 5 frames in 5 seconds over XWayland (Xeyes also reacts to XWayland windows in the monitor).

* Comment:

 I'm thinking this is an issue with the monitor not properly communicating with sway/wlroots, as this doesn't happen with other configurations. The only way to fix the issue is to exit and restart sway (restarting the computer isn't required). I'm not sure if it says so in the log, but when I exit sway it says:  
[common/ipc-client.c:88] Unable to receive IPC responseminor), code Window (no extension), sequence 22785, value 98566145
[common/ipc-client.c:88] Unable to receive IPC response

The monitor is an LG 24MP58VQ-P 1920x1080p @ 60Hz. Let me know if you need any more info.

**Comment by FoxtrotFaux** [comment]
(This might be a duplicate of #5245.)

**Comment by emersion** [comment]
Your bug report is missing debug logs. The one you provided has been generated with the Wayland backend.

**Comment by FoxtrotFaux** [comment]
Oh, sorry, I just did it according to the issue template. Is that just the log printed to the tty when starting it there? I couldn't find anything in the man pages about the DRM backend, am I missing something?

**Comment by RedSoxFan** [comment]
> Oh, sorry, I just did it according to the issue template. Is that just the log printed to the tty when starting it there?

Yes, obtain the debug log by starting Sway from a TTY. The log attached above is obtained from running Sway inside of Sway. Just use the same command (`sway -d 2> ~/sway.log`), but from a TTY instead of inside Sway.

>  I couldn't find anything in the man pages about the DRM backend, am I missing something?

The DRM (Direct Rendering Mancger) backend uses the physical hardware. If you start Sway from a TTY, it will use the DRM backend. When you start Sway inside of Sway, it will use the Wayland backend and behave the same as any other application would.

**Comment by FoxtrotFaux** [comment]
* New Debug Log: https://gist.github.com/FoxtrotFaux/123ae5b902bb1f13985c62ed7980bd80

**Comment by emersion** [comment]
From Sway point-of-view it seems like nothing goes wrong. Is there anything interesting in `dmesg`?

**Comment by FoxtrotFaux** [comment]
The CRTC of the HDMI monitor changes from 47 on first connect to -1 on reconnect. Not sure if that's correct. I'll post dmesg in next comment.

**Comment by FoxtrotFaux** [comment]
No errors or anything of interest in dmesg as far as I can tell.
* Dmesg: https://gist.github.com/FoxtrotFaux/68dba3ef1a4f376c960390cf4c99a009

**Comment by emersion** [comment]
Grepping for `'HDMI-A-1' connected`, CRTC 5 is used on reconnect:

```
2020-04-29 01:58:25 - [backend/drm/drm.c:1106] Reallocating CRTCs
2020-04-29 01:58:25 - [backend/drm/drm.c:1117] State before reallocation:
2020-04-29 01:58:25 - [backend/drm/drm.c:1123]   'DP-1' crtc=-1 state=0 desired_enabled=0
2020-04-29 01:58:25 - [backend/drm/drm.c:1123]   'DP-2' crtc=-1 state=0 desired_enabled=0
2020-04-29 01:58:25 - [backend/drm/drm.c:1123]   'DP-3' crtc=-1 state=0 desired_enabled=0
2020-04-29 01:58:25 - [backend/drm/drm.c:1123]   'HDMI-A-1' crtc=-1 state=1 desired_enabled=1
2020-04-29 01:58:25 - [backend/drm/drm.c:1123]   'DVI-D-1' crtc=1 state=3 desired_enabled=1
2020-04-29 01:58:25 - [backend/drm/drm.c:1175] State after reallocation:
2020-04-29 01:58:25 - [backend/drm/drm.c:1182]   'DP-1' crtc=-1 state=0 desired_enabled=0
2020-04-29 01:58:25 - [backend/drm/drm.c:1182]   'DP-2' crtc=-1 state=0 desired_enabled=0
2020-04-29 01:58:25 - [backend/drm/drm.c:1182]   'DP-3' crtc=-1 state=0 desired_enabled=0
2020-04-29 01:58:25 - [backend/drm/drm.c:1182]   'HDMI-A-1' crtc=5 state=1 desired_enabled=1
2020-04-29 01:58:25 - [backend/drm/drm.c:1182]   'DVI-D-1' crtc=1 state=3 desired_enabled=1
```

**Comment by FoxtrotFaux** [comment]
Ah, I was just skimming. Wasn't sure if it was significant.

**Comment by FoxtrotFaux** [comment]
Should I chalk this issue up to fussy monitor and close it? I haven't found any new information about where this issue is coming from.

**Comment by dinnymate** [comment]
I'started to experience this after updating sway. It was working fine before, so not a monitor issue. Also works fine in gdm and other DEs.

**Comment by matthewwardrop** [comment]
@FoxtrotFaux I experience this fairly reliably as well on the latest stable versions of wlroots and sway (not the git versions, which do not work at all for me). Did you ever sort it out?

**Comment by FoxtrotFaux** [comment]
> @FoxtrotFaux I experience this fairly reliably as well on the latest stable versions of wlroots and sway (not the git versions, which do not work at all for me). Did you ever sort it out?

No, I did not. My setup changed to laptops as I was mostly on the road for a while. I should be getting a monitor at my station soon, so I'll take another stab with fresh eyes.
